client/components/settingPage/index.js

```diff
@@ -1,9 +1,250 @@
-import React from 'react';
+import axios from 'axios';
+import { Alert } from 'react-bootstrap';
+import { Button } from 'react-bootstrap';
+import React, { useState, useRef, useContext } from 'react';
+import { firebaseStorage } from '../../../Docs/firebase-config/firebase';
+
+import './setting.css';
+
+export const SettingPage = ({ user }) => {
+  const [curTab, setCurTab] = useState('profile');
+  const [msg, setMsg] = useState('');
+  const [error, setError] = useState('');
+  const [usrEmail, setUsrEmail] = useState('');
+  const [usrName, setUsrName] = useState('');
+  const [usrOldPwd, setUsrOldPwd] = useState('');
+  const [usrNewPwd, setUsrNewPwd] = useState('');
+  const [ccNum, setCcNum] = useState(null);
+  const [ccHolder, setCcHolder] = useState(null);
+  const [ccDate, setCcDate] = useState(null);
+  const [ccCVC, setCcCVC] = useState(null);
+
+  const fileRef = useRef();
+
+  const newImg = () => {
+    document.getElementById('hiddenImg').click();
+  };
+
+  const sendImg = () => {
+    console.log(fileRef.current.files[0]);
+    const storageRef = firebaseStorage.ref(fileRef.current.files[0].name);
+    storageRef
+      .put(fileRef.current.files[0])
+      .then((snap) => {
+        return snap.ref.getDownloadURL();
+      })
+      .then((url) => {
+        document.getElementById('profilePic').src = url;
+        return axios.put('/user/info', {
+          userId: user._id,
+          photoUrl: url,
+        });
+      })
+      .catch((err) => console.log(err));
+  };
+
+  const submitHandler = async (e) => {
+    e.preventDefault();
+    setError('');
+    setMsg('');
+
+    try {
+      if (!usrOldPwd) {
+        await axios.put('/user/info', {
+          userId: user._id,
+          username: usrName || undefined,
+          email: usrEmail || undefined,
+        });
+        setMsg('Data updated');
+      } else {
+        axios
+          .put('/user/info', {
+            userId: user._id,
+            username: usrName || undefined,
+            email: usrEmail || undefined,
+            password: usrOldPwd,
+            newPassword: usrNewPwd,
+          })
+          .then(() => {
+            setMsg('Data updated');
+          })
+          .catch(() => setError('Invalid passwords'));
+      }
+      console.log('UPDATED');
+    } catch (error) {
+      setError('ERROR: ' + error);
+    }
+  };
+
+  const paymentHandler = (e) => {
+    e.preventDefault();
+
+    axios
+      .post('/user/payment', {
+        userId: user._id,
+        cardNumber: ccNum,
+        cardHolder: ccHolder,
+        cardExpiredDate: ccDate,
+      })
+      .then(() => setMsg('Payment Updated'))
+      .catch(() => setError('Unable to update payment'));
+  };
 
-export const SettingPage = () => {
   return (
     <>
-      <h1>Setting Page</h1>
+      {user && (
```

Comment:
I would consider modularizing the code below. Thats ~150 lines of code in one conditional, and would allow a better separation of concerns for your different tabs, if that is what I am looking at. Also quite hard to tell where the conditionals start and end.

Reply:
Okay

Reply:
We decided refactor later, resolving conversation